python/tests/kernel/test_call_return.py

```diff
@@ -0,0 +1,70 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+import cudaq
+
+
+def test_call_with_return_bool():
+
+    @cudaq.kernel()
+    def callee(q: cudaq.qubit) -> bool:
+        h(q)
+        m = mz(q)
+        return m
+
+    @cudaq.kernel()
+    def caller() -> bool:
```

Comment:
Q: I could be mistaken, or it could be a typo, but the example in issue #2250, makes it look like the caller function does not need to specify the return type hint?

Reply:
The simpler test in the issue just checks for crashes, but this one should verify that the execution is also correct. Added both, thanks!